### PR TITLE
Code refactoring and boundary renumbering

### PR DESCRIPTION
**Pull request by jzwar**
# Reorder Boundary
Reorder according to #78 

## Addressed issues
*  #78 

## Checklists
* [ ] Documentations are up-to-date.
* [ ] Added example(s)
* [ ] Added test(s)

### CODE REVIEW

**Comment by j042** [comment]
I think you have to inline that func. the func with umgekehrt name

**Comment by j042** [comment]
thank you for inlining that func

File: cpp/splinepy/py/py_spline.hpp [hunk 1]
```diff
@@ -844,11 +844,43 @@ inline py::list ExtractBezierPatches(const PySpline& spline) {
 }
 
 /// boundary spline extraction
-inline PySpline ExtractBoundary(const PySpline& spline,
-                                const int& boundary_normal_axis,
-                                const int& extrema) {
-  return PySpline(
-      spline.Core()->SplinepyExtractBoundary(boundary_normal_axis, extrema));
+inline py::list ExtractBoundaries(const PySpline& spline,
```
Comment (j042) [inline cpp/splinepy/py/py_spline.hpp:847]: nice

File: cpp/splinepy/py/py_spline.hpp [hunk 2]
```diff
@@ -844,11 +844,43 @@ inline py::list ExtractBezierPatches(const PySpline& spline) {
...
+
+  return boundary_splines;
+}
+
+/// boundary spline extraction from axis and extreme
+inline PySpline ExtractBoundaryFromAxisAndExtrema(const PySpline& spline,
+                                                  const int& axis,
+                                                  const int& extreme) {
+  // Determine corresponding ID
+  const int boundary_id = (extreme > 0) ? 2 * axis + 1 : 2 * axis;
```
Comment (j042) [inline cpp/splinepy/py/py_spline.hpp:880]: I'd love this helper for helper, maybe for next PR

**Review by j042 — APPROVED** [review]
awesome

File: cpp/splinepy/py/py_spline.hpp (hunk 2 above)
Comment (jzwar) [reply cpp/splinepy/py/py_spline.hpp:880]: Next PR <3

**Pull request merged**